crates/libs/core/src/handles.rs

```diff
@@ -0,0 +1,45 @@
+/// Custom code to free a handle.
+///
+/// This is similar to the `Drop` trait, and may be used to implement `Drop`, but allows handles
+/// to be dropped depending on context.
+pub trait Free {
+    /// Calls the handle's free function.
+    ///
+    /// # Safety
+    /// The handle must be owned by the caller and safe to free.    
+    unsafe fn free(&mut self);
+}
+
+/// A wrapper to provide ownership for handles to automatically drop via the handle's `Free` trait.
+#[repr(transparent)]
+#[derive(Clone, PartialEq, Eq, Default, Debug)]
```

Comment:
I strongly doubt `Owned` should implement `Clone`, at least not safely, because this way you could safely drop the same handle twice because `unsafe fn free()` is hidden away in `drop()`.

Reply:
Wow, that was really poor timing with the merge.

Reply:
See #3014